### DISCUSSION TITLE
.env file not getting pushed to servers

### DISCUSSION THREAD

**Discussion by Albi-Oxa** [basecamp/kamal#929] [Q&A]
Hello, I have build a simple nextjs app for trying to understand kamal better before I use it in production. What I need to do is that I have variables in my code that are taken form the .env file. Locally it runs perfectly but I want to use them also when kamal deploys this app.
### Steps I have followed
Fist created the nextjs app using one the free templates that nextjs offers. Then initialized kamal and git in this folder. Now the `kamal init` command created files like in the documentation, one of them being .env. Now inside this .env I put the variables that kamal generated and the variables I use in my code. Then committed the repository and ran `kamal setup` command. Everything went smoothly and deployment was successful. When I open the page in my web browser where I have used the variable that I added, the value of that isn't retrived. 
### What I have tried 
I read the documentation and tried to use the `kamal env push` but it doesn't do anything. Tried to add the env block in the `deploy.yml` like this:
``` 
env:
  secret:
    - VALUE
 ```
and ran `kamal env push` but not working either. Also tried with `kamal deploy` but no luck either. How I use this variable VALUE is simple, as I said I am doing it just for testing purposes, like this:
```
import React from 'react'

function page() {
  const envVar = process.env.VALUE

  return (
    <div>
      <div>{envVar}</div>
      <div>End of ENV</div>
    </div>
  )
}

export default page
```
 
Would appreciate any help. Maybe i am missing something or doing it wrong but I am new to programming and the kamal tool and would like to really learn it. If you have any question please do ask me.Thank you! 
Also if you have please do provide and tips or tricks to keep in mind when configuring the `deploy.yml`

**Comment by jericopulvera** [comment]
I ran into this issue as well using Kamal 2. It looks like Next.js requires a .env file inside the container, and the secrets from Kamal are only supplied at Docker runtime, so it doesn't work.

I mentioned the solution I came up with here https://github.com/basecamp/kamal/issues/925#issuecomment-2408655462.

The issue with this currently is that I have to keep changing my .env before deploying, which is annoying.

I was thinking of adding a pre-build step where it copies the .kamal/secrets to a .env file.

**Reply by Derick1530** [reply]
@jericopulvera  did you make it work? if yes can you please share your solution ?
